### PR TITLE
Do not wait when there is not log entry.

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Bug Fix

Short description (up to few sentences):

https://github.com/yandex/ClickHouse/issues/3611

Empty ReplicatedMergeTree table breaks optimize final.

Detailed description (optional):

...

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
A test?

File: dbms/src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -3015,6 +3015,7 @@ bool StorageReplicatedMergeTree::optimize(const ASTPtr & query, const ASTPtr & p
     }
 
     ReplicatedMergeTreeLogEntryData merge_entry;
+    bool need_wait = false;
```
Comment (alexey-milovidov) [inline dbms/src/Storages/StorageReplicatedMergeTree.cpp]: Or we can check `merge_entry` directly, without this flag.

**Comment by amosbird** [comment]
updated. Not sure if this fixes the original issue but at least it covers what @den-crane had reproduced.

**Comment by alexey-milovidov** [comment]
Ok.

PS. It seems difficult to provide a good test.
(Integration test is possible but not worth.)

**Comment by amosbird** [comment]
oops, do you mind revert this. I just pushed a refinement...

**Pull request merged**